modules/react/pagination/lib/Pagination/Controls.tsx

```diff
@@ -1,115 +1,138 @@
 import * as React from 'react';
-import {TertiaryButton, TertiaryButtonProps} from '@workday/canvas-kit-react/button';
+import {TertiaryButton} from '@workday/canvas-kit-react/button';
+import {createComponent, ExtractProps} from '@workday/canvas-kit-react/common';
 import {
   chevronLeftSmallIcon,
   chevron2xLeftSmallIcon,
   chevronRightSmallIcon,
   chevron2xRightSmallIcon,
 } from '@workday/canvas-system-icons-web';
 
-import {PaginationModel} from './types';
 import {HStack} from '@workday/canvas-kit-react/layout';
 import {useRTL} from './common/utils/useRTL';
+import {PaginationContext} from './usePaginationModel';
 
-export type ControlButtonProps = TertiaryButtonProps &
-  React.ButtonHTMLAttributes<HTMLButtonElement> & {
-    model: PaginationModel;
-  };
+export interface ControlButtonProps extends ExtractProps<typeof TertiaryButton, never> {
+  onClick?: React.MouseEventHandler<HTMLButtonElement>;
+}
 
-export type ControlsProps = React.HTMLAttributes<HTMLDivElement>;
+export interface PaginationControlsProps {}
```

Comment:
The change is additive. `Pagination.Controls` is not loosing `React.HTMLAttributes<HTMLDivElement>` because it is getting it from `createComponent('div')`. So all the props that worked before will still be valid.

`Pagination.Controls` gains the `as` and `ref` props.

Reply:
<img width="725" alt="image" src="https://user-images.githubusercontent.com/338257/222586505-f3068352-3d77-4115-a86f-658cf8293dac.png">